client/comms: server not listening when client connection is attempted

After merge of PR  , I occasionally get errors in CI and on my local machine at https://github.com/decred/dcrdex/blob/master/client/comms/wsconn_test.go#L227

`TestWsConn: wsconn_test.go:227: failed to get ErrCertRequired for no cert connection`

I supplemented the error and it seems that the cause is the server not yet listening when the connection attempt is made:

```
 $   go test -v -short -race -count 1 -tags pgonline ./client/comms/...
=== RUN   TestWsConn
2020-05-25 11:15:59.368 [INF] Debug: Generating TLS certificates...
2020-05-25 11:15:59.446 [INF] Debug: Done generating TLS certificates
    TestWsConn: wsconn_test.go:227: failed to get ErrCertRequired for no cert connection, got dial tcp 127.0.0.1:6060: connect: connection refused
http: Server closed
--- FAIL: TestWsConn (0.11s)
FAIL
FAIL	decred.org/dcrdex/client/comms	0.122s
FAIL
```

This is because the server is started in a goroutine, and a connection attempt is made immediately after.  This demonstrates how we should never assume the code inside a goroutine is run immediately:

https://github.com/decred/dcrdex/blob/1bc5777289db4d612521e18b7ddfd3c30b6d51fa/client/comms/wsconn_test.go#L197-L203 

We need to wait for ListenAndServer to actually start listening.

--- a/client/comms/wsconn_test.go
+++ b/client/comms/wsconn_test.go
@@ -8,6 +8,7 @@ import (
 	"errors"
 	"fmt"
 	"io/ioutil"
+	"net"
 	"net/http"
 	"os"
 	"sync"
@@ -194,18 +195,28 @@ func TestWsConn(t *testing.T) {
 	defer server.Shutdown(context.Background())
 
 	wg.Add(1)
-	serverReady := make(chan struct{}, 1)
+	serverReady := make(chan error, 1)
 	go func() {
 		defer wg.Done()
-		serverReady <- struct{}{}
-		err := server.ListenAndServeTLS(certFile.Name(), keyFile.Name())
+
+		ln, err := net.Listen("tcp", server.Addr)
+		serverReady <- err
+		if err != nil {
+			return
+		}
+		defer ln.Close()
+
+		err = server.ServeTLS(ln, certFile.Name(), keyFile.Name())
 		if err != nil {
 			fmt.Println(err)
 		}
 	}()
 
 	// wait for server to start listening before connecting
-	<-serverReady
+	err = <-serverReady
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	setupWsConn := func(cert []byte) (*wsConn, error) {
 		cfg := &WsCfg{

--- a/client/comms/wsconn_test.go
+++ b/client/comms/wsconn_test.go
@@ -194,13 +194,18 @@ func TestWsConn(t *testing.T) {
 	defer server.Shutdown(context.Background())
 
 	wg.Add(1)
+	serverReady := make(chan struct{}, 1)
 	go func() {
 		defer wg.Done()
+		serverReady <- struct{}{}
 		err := server.ListenAndServeTLS(certFile.Name(), keyFile.Name())
 		if err != nil {
 			fmt.Println(err)
 		}
 	}()
+
+	// wait for server to start listening before connecting
+	<-serverReady
 
 	setupWsConn := func(cert []byte) (*wsConn, error) {
 		cfg := &WsCfg{

